### ISSUE TITLE
Support for generic casts

### ISSUE THREAD

**Issue by thekid** [xp-lang/xp-generics#2]
Test:

```php
public function generic_cast() {
  $r= $this->run('class <T><E> {
    public function run($args) {
      return (E)$args;
    }
  }', 10);

  Assert::equals('10', $r);
}
```

**Comment by thekid** [comment]
Given the following cast expression: `(E)$arg`, this is what happens:

```php
// 1: Find out whether containing class is generic, e.g. T<string>
if ($generic= \xp::$meta[\xp::$cn[self::class]]['class'][DETAIL_GENERIC] ?? null) {

  // 2. Find whether the generic definition (e.g. #[Generic(self: 'K, V')]) contains the type
  $contained= array_search('E', preg_split(
    '/, ?/',
    \xp::$meta[$generic[0]]['class'][DETAIL_ANNOTATIONS]['generic']['self']
  ));
  
  // 3. E refers to a generic component, use the associated argument to cast
  if (false !== $contained) return $generic[1][$contained]->cast($arg);
}

// 4. Otherwise, use the type as-is
return cast('E', $arg);
```

**Comment by thekid** [comment]
Implemented in #6
